docs/source/peers/peers.md

```diff
@@ -369,61 +447,79 @@ can be accepted by the network. This is literally what it means to achieve
+In response to inconsistent transaction responses, caused either by
+nondeterministic chaincode or any other issue, the gateway service terminates
+the transaction workflow and returns a failed proposal message to the client. The
+client can then resubmit the transaction proposal or an amended transaction proposal.
+
+If the transaction proposal is endorsed, at the end of phase 1, the gateway service
+returns a successful proposal message to the client and submits the transaction to
+the ordering service, as described next in phase 2.
 
 ### Phase 2: Ordering and packaging transactions into blocks
 
-The second phase of the transaction workflow is the packaging phase. The orderer
-is pivotal to this process --- it receives transactions containing endorsed
-transaction proposal responses from many applications, and orders the
-transactions into blocks. For more details about the
-ordering and packaging phase, check out our
-[conceptual information about the ordering phase](../orderer/ordering_service.html#phase-two-ordering-and-packaging-transactions-into-blocks).
+The second phase of the transaction workflow is the ordering and packaging phase. The
+ordering service (orderer nodes) receives transactions containing signed and endorsed
+proposal responses, from one or more applications via the gateway service, and orders
+and packages the transactions into blocks. For details about the ordering and packaging
+phase, see the [conceptual information about the ordering phase](../orderer/ordering_service.html#phase-two-ordering-and-packaging-transactions-into-blocks).
 
 ### Phase 3: Validation and commit
 
-At the end of phase 2, we see that orderers have been responsible for the simple
-but vital processes of collecting proposed transaction updates, ordering them,
-and packaging them into blocks, ready for distribution to the peers.
+At the end of phase 2, we see that orderers are responsible for the simple
+but vital processes of ordering endorsed transactions forwarded by the gateway
+service and packaging them into blocks for distribution to the peers.
 
-The final phase of the transaction workflow involves the distribution and
-subsequent validation of blocks from the orderer to the peers, where they can be
+The final phase of the transaction workflow involves the distribution (and
+subsequent validation) of blocks from the orderer to the peers, where they are
 committed to the ledger. Specifically, at each peer, every transaction within a
 block is validated to ensure that it has been consistently endorsed by all
-relevant organizations before it is committed to the ledger. Failed transactions
-are retained for audit, but are not committed to the ledger.
+relevant organizations before being committed to the ledger. A different
+validation result across on any peers means endorsements by one or more of  
+those peers are not equivalent. Failed transactions are retained for audit,
```

Comment:
I would remove this.
The peer validation phase does check that endorsements match, and you can mention this. But note that gateway will not even submit the transaction if they don't match, so mentioning this will confuse people.
Also "different validation result" is not correct here... you mean "different execution result".
Finally, note that there are other potential reasons for transaction invalidation besides executions/endorsements not matching.

Reply:
yes so there is a submit by client during proposal phase to the service and then another submit by the service apparently. will look to differentiate. 
will diff the execution/simulation/run vs. validation. they do have to run it to validate it though afaik.

Reply:
Seems there's a submit by the app to the gateway service, of a proposal, and then a submit of by the service to the orderers, for an endorsed transaction. Hence good to distinguish the initial submit by calling it a send instead of submit, if that works.

Reply:
Andy uses submit so to be consistent will use submit ... the submit "to" should clarify it enough.